bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/handler/NooliteMTRF64BridgeHandler.java

```diff
@@ -0,0 +1,206 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Petr Shatsillo - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class NooliteMTRF64BridgeHandler extends BaseBridgeHandler {
+
+    private static Logger logger = LoggerFactory.getLogger(NooliteMTRF64BridgeHandler.class);
+    @Nullable
+    static NooliteMTRF64Adapter adapter;
+    @Nullable
+    private NooliteBridgeConfiguration bridgeConfig;
+    public static Map<String, NooliteHandler> thingHandlerMap = new HashMap<String, NooliteHandler>();
+    @Nullable
+    static NooliteHandler nooliteHandler;
+
+    public NooliteMTRF64BridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Close bridge");
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Initializing...");
+        bridgeConfig = getConfigAs(NooliteBridgeConfiguration.class);
+
+        // logger.debug("Checking Noolite adapter connection. {}", thing.getStatus());
+        if (thing.getStatus() != ThingStatus.ONLINE) {
+            connect();
+        }
+    }
+
+    @SuppressWarnings("null")
+    private void connect() {
+        logger.debug("Connecting to Noolite adapter");
+        try {
+            if (bridgeConfig.serial != null) {
+                if (adapter == null) {
+                    adapter = new NooliteMTRF64Adapter();
+                }
+            }
+            if (adapter != null) {
+                adapter.disconnect();
+                adapter.connect(bridgeConfig);
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Com-port error");
+            logger.debug(e.getMessage());
+        }
+    }
+
+    @SuppressWarnings({ "null", "unused" })
+    public void registerMegadThingListener(NooliteHandler thingHandler) {
+        String thingID = thingHandler.getThing().getConfiguration().get("type").toString() + "."
+                + (Integer.parseInt(thingHandler.getThing().getConfiguration().get("port").toString()));
+        if (thingHandlerMap.get(thingID) != null) {
+            thingHandlerMap.remove(thingID);
+        }
+        logger.debug("thingHandler for thing: '{}'", thingID);
+        if (thingHandlerMap.get(thingID) == null) {
```

Comment:
If you first remove the thingID on line 113 than this test on line 116 will always be true.